The job `x86_64-gnu-llvm-6.0` of your PR [failed](https://dev.azure.com/rust-lang/rust/_build/results?buildId=7460&view=logs&jobId=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e) ([raw log](https://rust-lang.github.io/rust-log-analyzer/log-viewer/#https://dev.azure.com/rust-lang/e71b0ddf-dd27-435a-873c-e30f86eea377/_apis/build/builds/7460/logs/49)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
2019-09-05T20:19:04.7171377Z ##[command]git remote add origin https://github.com/rust-lang/rust
2019-09-05T20:19:04.7353899Z ##[command]git config gc.auto 0
2019-09-05T20:19:04.7437824Z ##[command]git config --get-all http.https://github.com/rust-lang/rust.extraheader
2019-09-05T20:19:04.7485915Z ##[command]git config --get-all http.proxy
2019-09-05T20:19:04.7634432Z ##[command]git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules --depth=2 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/64192/merge:refs/remotes/pull/64192/merge
---
2019-09-05T21:27:27.8087989Z .................................................................................................... 1500/9000
2019-09-05T21:27:34.4611529Z .................................................................................................... 1600/9000
2019-09-05T21:27:48.7120820Z ......................................................i...............i............................. 1700/9000
2019-09-05T21:27:57.6490683Z .................................................................................................... 1800/9000
2019-09-05T21:28:14.1088850Z .............................................iiiii.................................................. 1900/9000
2019-09-05T21:28:26.3934954Z .................................................................................................... 2100/9000
2019-09-05T21:28:29.2833601Z .................................................................................................... 2200/9000
2019-09-05T21:28:33.3348362Z .................................................................................................... 2300/9000
2019-09-05T21:28:42.2045413Z .................................................................................................... 2400/9000
---
2019-09-05T21:32:01.3190662Z .................................i...............i.................................................. 4700/9000
2019-09-05T21:32:14.1949029Z .................................................................................................... 4800/9000
2019-09-05T21:32:21.3653923Z .................................................................................................... 4900/9000
2019-09-05T21:32:33.4509761Z .................................................................................................... 5000/9000
2019-09-05T21:32:40.1941179Z ...............ii.ii................................................................................ 5100/9000
2019-09-05T21:32:52.2983302Z .................................................................................................... 5300/9000
2019-09-05T21:33:03.6612621Z ..............................................................................i..................... 5400/9000
2019-09-05T21:33:12.0970636Z .................................................................................................... 5500/9000
2019-09-05T21:33:18.8839205Z .................................................................................................... 5600/9000
2019-09-05T21:33:18.8839205Z .................................................................................................... 5600/9000
2019-09-05T21:33:30.5906683Z ........................................................................ii...i..ii...........i...... 5700/9000
2019-09-05T21:33:58.7177777Z .................................................................................................... 5900/9000
2019-09-05T21:34:09.7005655Z .................................................................................................... 6000/9000
2019-09-05T21:34:09.7005655Z .................................................................................................... 6000/9000
2019-09-05T21:34:20.3779197Z ..........................................................................i..ii..................... 6100/9000
2019-09-05T21:34:53.5723941Z .................................................................................................... 6300/9000
2019-09-05T21:34:55.8879656Z ..............................i..................................................................... 6400/9000
2019-09-05T21:34:58.3356422Z .................................................................................................... 6500/9000
2019-09-05T21:35:01.2584527Z .i.................................................................................................. 6600/9000
---
2019-09-05T21:40:19.9897911Z  finished in 21.251
2019-09-05T21:40:20.0114041Z Check compiletest suite=codegen mode=codegen (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-09-05T21:40:20.1887655Z 
2019-09-05T21:40:20.1887918Z running 149 tests
2019-09-05T21:40:23.8259493Z i....iii......iii..iiii....i.............................i..i..................i....i.........ii.i.i 100/149
2019-09-05T21:40:26.0345812Z ..iiii..............i.........iii.i......ii......
2019-09-05T21:40:26.0347716Z 
2019-09-05T21:40:26.0352865Z  finished in 6.024
2019-09-05T21:40:26.0596340Z Check compiletest suite=codegen-units mode=codegen-units (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-09-05T21:40:26.2295072Z 
---
2019-09-05T21:40:28.5422105Z  finished in 2.485
2019-09-05T21:40:28.5637331Z Check compiletest suite=assembly mode=assembly (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-09-05T21:40:28.7462007Z 
2019-09-05T21:40:28.7462273Z running 9 tests
2019-09-05T21:40:28.7463117Z iiiiiiiii
2019-09-05T21:40:28.7463536Z 
2019-09-05T21:40:28.7463580Z  finished in 0.182
2019-09-05T21:40:28.7686866Z Check compiletest suite=incremental mode=incremental (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-09-05T21:40:29.5386465Z 
---
2019-09-05T21:40:48.7683290Z  finished in 20.000
2019-09-05T21:40:48.7912255Z Check compiletest suite=debuginfo mode=debuginfo-gdb+lldb (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-09-05T21:40:48.9887117Z 
2019-09-05T21:40:48.9887385Z running 123 tests
2019-09-05T21:41:15.6304507Z .iiiii...i.....i..i...i..i.i.i..i.ii..i.i.....i..i....ii..........iiii..........i...ii...i.......ii. 100/123
2019-09-05T21:41:20.8283576Z i.i.i......iii.i.....ii
2019-09-05T21:41:20.8285356Z 
2019-09-05T21:41:20.8285576Z  finished in 32.037
2019-09-05T21:41:20.8295460Z Uplifting stage1 rustc (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
2019-09-05T21:41:20.8296829Z Copying stage2 rustc from stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu / x86_64-unknown-linux-gnu)
---
2019-09-05T21:42:31.3890489Z status: exit code: 101
2019-09-05T21:42:31.3891073Z command: "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/pprust-expr-roundtrip/a"
2019-09-05T21:42:31.3891133Z stdout:
2019-09-05T21:42:31.3891366Z ------------------------------------------
2019-09-05T21:42:31.3891430Z printed: box box x
2019-09-05T21:42:31.3891472Z printed: box x()
2019-09-05T21:42:31.3891524Z printed: box x.x(x)
2019-09-05T21:42:31.3891565Z printed: box x.x(x)
2019-09-05T21:42:31.3891624Z printed: box (x + x)
2019-09-05T21:42:31.3891666Z printed: box (x + x)
2019-09-05T21:42:31.3891706Z printed: box (x * x)
2019-09-05T21:42:31.3891761Z printed: box (x * x)
2019-09-05T21:42:31.3891801Z printed: box (x << x)
2019-09-05T21:42:31.3891842Z printed: box (x << x)
2019-09-05T21:42:31.3891900Z printed: box (x && x)
2019-09-05T21:42:31.3891940Z printed: box (x && x)
2019-09-05T21:42:31.3891980Z printed: box (x || x)
2019-09-05T21:42:31.3892018Z printed: box (x || x)
2019-09-05T21:42:31.3892076Z printed: box (x < x)
2019-09-05T21:42:31.3892115Z printed: box (x < x)
2019-09-05T21:42:31.3892402Z printed: box *x
2019-09-05T21:42:31.3892455Z printed: box if x { }
2019-09-05T21:42:31.3892662Z printed: box (move || x)
2019-09-05T21:42:31.3892704Z printed: box (x = x)
2019-09-05T21:42:31.3892759Z printed: box (x = x)
2019-09-05T21:42:31.3892801Z printed: box x.f
2019-09-05T21:42:31.3892841Z printed: box (x..x)
2019-09-05T21:42:31.3892991Z printed: box (x..x)
2019-09-05T21:42:31.3893054Z printed: box &x
2019-09-05T21:42:31.3893094Z printed: box (return)
2019-09-05T21:42:31.3893135Z printed: box (return x)
2019-09-05T21:42:31.3893191Z printed: box S{..x}
2019-09-05T21:42:31.3893231Z printed: box x?
2019-09-05T21:42:31.3893272Z printed: box let _ = x
2019-09-05T21:42:31.3893313Z printed: (box x)()
2019-09-05T21:42:31.3893371Z printed: x()()
2019-09-05T21:42:31.3893412Z printed: x.x(x)()
2019-09-05T21:42:31.3893452Z printed: x.x(x)()
2019-09-05T21:42:31.3893503Z printed: (x + x)()
2019-09-05T21:42:31.3893543Z printed: (x + x)()
2019-09-05T21:42:31.3893583Z printed: (x * x)()
2019-09-05T21:42:31.3893631Z printed: (x * x)()
2019-09-05T21:42:31.3893680Z printed: (x << x)()
2019-09-05T21:42:31.3893720Z printed: (x << x)()
2019-09-05T21:42:31.3893760Z printed: (x && x)()
2019-09-05T21:42:31.3894199Z printed: (x && x)()
2019-09-05T21:42:31.3894243Z printed: (x || x)()
2019-09-05T21:42:31.3894284Z printed: (x || x)()
2019-09-05T21:42:31.3894347Z printed: (x < x)()
2019-09-05T21:42:31.3894389Z printed: (x < x)()
2019-09-05T21:42:31.3894430Z printed: (*x)()
2019-09-05T21:42:31.3894471Z printed: if x { }()
2019-09-05T21:42:31.3894522Z printed: (move || x)()
2019-09-05T21:42:31.3894564Z printed: (x = x)()
2019-09-05T21:42:31.3894605Z printed: (x = x)()
2019-09-05T21:42:31.3894655Z printed: (x.f)()
2019-09-05T21:42:31.3894696Z printed: (x..x)()
2019-09-05T21:42:31.3894738Z printed: (x..x)()
2019-09-05T21:42:31.3894778Z printed: (&x)()
2019-09-05T21:42:31.3894829Z printed: (return)()
2019-09-05T21:42:31.3894871Z printed: (return x)()
2019-09-05T21:42:31.3894913Z printed: S{..x}()
2019-09-05T21:42:31.3894970Z printed: x?()
2019-09-05T21:42:31.3895013Z printed: (let _ = x)()
2019-09-05T21:42:31.3895055Z printed: (box x).x(x)
2019-09-05T21:42:31.3895106Z printed: x().x(x)
2019-09-05T21:42:31.3895148Z printed: x.x(x).x(x)
2019-09-05T21:42:31.3895190Z printed: x.x(x).x(x)
2019-09-05T21:42:31.3895231Z printed: (x + x).x(x)
2019-09-05T21:42:31.3895298Z printed: (x + x).x(x)
2019-09-05T21:42:31.3895340Z printed: (x * x).x(x)
2019-09-05T21:42:31.3895381Z printed: (x * x).x(x)
2019-09-05T21:42:31.3895440Z printed: (x << x).x(x)
2019-09-05T21:42:31.3895483Z printed: (x << x).x(x)
2019-09-05T21:42:31.3895524Z printed: (x && x).x(x)
2019-09-05T21:42:31.3895566Z printed: (x && x).x(x)
2019-09-05T21:42:31.3895625Z printed: (x || x).x(x)
2019-09-05T21:42:31.3895666Z printed: (x || x).x(x)
2019-09-05T21:42:31.3895707Z printed: (x < x).x(x)
2019-09-05T21:42:31.3895762Z printed: (x < x).x(x)
2019-09-05T21:42:31.3895804Z printed: (*x).x(x)
2019-09-05T21:42:31.3895845Z printed: if x { }.x(x)
2019-09-05T21:42:31.3895903Z printed: (move || x).x(x)
2019-09-05T21:42:31.3895954Z printed: (x = x).x(x)
2019-09-05T21:42:31.3895995Z printed: (x = x).x(x)
2019-09-05T21:42:31.3896037Z printed: x.f.x(x)
2019-09-05T21:42:31.3896096Z printed: (x..x).x(x)
2019-09-05T21:42:31.3896137Z printed: (x..x).x(x)
2019-09-05T21:42:31.3896178Z printed: (&x).x(x)
2019-09-05T21:42:31.3896243Z printed: (return).x(x)
2019-09-05T21:42:31.3896286Z printed: (return x).x(x)
2019-09-05T21:42:31.3896329Z printed: S{..x}.x(x)
2019-09-05T21:42:31.3896370Z printed: x?.x(x)
2019-09-05T21:42:31.3896431Z printed: (let _ = x).x(x)
2019-09-05T21:42:31.3896474Z printed: x.x(box x)
2019-09-05T21:42:31.3896515Z printed: x.x(x())
2019-09-05T21:42:31.3896574Z printed: x.x(x.x(x))
2019-09-05T21:42:31.3896616Z printed: x.x(x.x(x))
2019-09-05T21:42:31.3896657Z printed: x.x(x + x)
2019-09-05T21:42:31.3896715Z printed: x.x(x + x)
2019-09-05T21:42:31.3896756Z printed: x.x(x * x)
2019-09-05T21:42:31.3896797Z printed: x.x(x * x)
2019-09-05T21:42:31.3896838Z printed: x.x(x << x)
2019-09-05T21:42:31.3897028Z printed: x.x(x << x)
2019-09-05T21:42:31.3897069Z printed: x.x(x && x)
2019-09-05T21:42:31.3897110Z printed: x.x(x && x)
2019-09-05T21:42:31.3897168Z printed: x.x(x || x)
2019-09-05T21:42:31.3897210Z printed: x.x(x || x)
2019-09-05T21:42:31.3897252Z printed: x.x(x < x)
2019-09-05T21:42:31.3897293Z printed: x.x(x < x)
2019-09-05T21:42:31.3897569Z printed: x.x(*x)
2019-09-05T21:42:31.3897615Z printed: x.x(if x { })
2019-09-05T21:42:31.3897656Z printed: x.x(move || x)
2019-09-05T21:42:31.3897709Z printed: x.x(x = x)
2019-09-05T21:42:31.3897749Z printed: x.x(x = x)
2019-09-05T21:42:31.3897788Z printed: x.x(x.f)
2019-09-05T21:42:31.3897846Z printed: x.x(x..x)
2019-09-05T21:42:31.3897886Z printed: x.x(x..x)
2019-09-05T21:42:31.3897925Z printed: x.x(&x)
2019-09-05T21:42:31.3897965Z printed: x.x(return)
2019-09-05T21:42:31.3898022Z printed: x.x(return x)
2019-09-05T21:42:31.3898062Z printed: x.x(S{..x})
2019-09-05T21:42:31.3898101Z printed: x.x(x?)
2019-09-05T21:42:31.3898161Z printed: x.x(let _ = x)
2019-09-05T21:42:31.3898210Z printed: box x + x
2019-09-05T21:42:31.3898249Z printed: x() + x
2019-09-05T21:42:31.3898289Z printed: x.x(x) + x
2019-09-05T21:42:31.3898346Z printed: x.x(x) + x
2019-09-05T21:42:31.3898385Z printed: x + x + x
2019-09-05T21:42:31.3898425Z printed: x + x + x
2019-09-05T21:42:31.3898482Z printed: x * x + x
2019-09-05T21:42:31.3898803Z printed: x * x + x
2019-09-05T21:42:31.3898924Z printed: (x << x) + x
2019-09-05T21:42:31.3898983Z printed: (x << x) + x
2019-09-05T21:42:31.3899022Z printed: (x && x) + x
2019-09-05T21:42:31.3899063Z printed: (x && x) + x
2019-09-05T21:42:31.3899102Z printed: (x || x) + x
2019-09-05T21:42:31.3899159Z printed: (x || x) + x
2019-09-05T21:42:31.3899198Z printed: (x < x) + x
2019-09-05T21:42:31.3899238Z printed: (x < x) + x
2019-09-05T21:42:31.3899291Z printed: *x + x
2019-09-05T21:42:31.3899502Z printed: if x { } + x
2019-09-05T21:42:31.3899550Z printed: (move || x) + x
2019-09-05T21:42:31.3899590Z printed: (x = x) + x
2019-09-05T21:42:31.3899654Z printed: (x = x) + x
2019-09-05T21:42:31.3899694Z printed: x.f + x
2019-09-05T21:42:31.3899734Z printed: (x..x) + x
2019-09-05T21:42:31.3899790Z printed: (x..x) + x
2019-09-05T21:42:31.3899829Z printed: &x + x
2019-09-05T21:42:31.3899869Z printed: (return) + x
2019-09-05T21:42:31.3899926Z printed: (return x) + x
2019-09-05T21:42:31.3899973Z printed: S{..x} + x
2019-09-05T21:42:31.3900130Z printed: x? + x
2019-09-05T21:42:31.3900240Z printed: (let _ = x) + x
2019-09-05T21:42:31.3900293Z printed: x + box x
2019-09-05T21:42:31.3900333Z printed: x + x()
2019-09-05T21:42:31.3900373Z printed: x + x.x(x)
2019-09-05T21:42:31.3900427Z printed: x + x.x(x)
2019-09-05T21:42:31.3900468Z printed: x + (x + x)
2019-09-05T21:42:31.3900508Z printed: x + (x + x)
2019-09-05T21:42:31.3900547Z printed: x + x * x
2019-09-05T21:42:31.3902905Z printed: x + x * x
2019-09-05T21:42:31.3902952Z printed: x + (x << x)
2019-09-05T21:42:31.3902993Z printed: x + (x << x)
2019-09-05T21:42:31.3903048Z printed: x + (x && x)
2019-09-05T21:42:31.3903102Z printed: x + (x && x)
2019-09-05T21:42:31.3903141Z printed: x + (x || x)
2019-09-05T21:42:31.3903198Z printed: x + (x || x)
2019-09-05T21:42:31.3903238Z printed: x + (x < x)
2019-09-05T21:42:31.3903278Z printed: x + (x < x)
2019-09-05T21:42:31.3903317Z printed: x + *x
2019-09-05T21:42:31.3903373Z printed: x + if x { }
2019-09-05T21:42:31.3903421Z printed: x + (move || x)
2019-09-05T21:42:31.3903462Z printed: x + (x = x)
2019-09-05T21:42:31.3903517Z printed: x + (x = x)
2019-09-05T21:42:31.3903557Z printed: x + x.f
2019-09-05T21:42:31.3903597Z printed: x + (x..x)
2019-09-05T21:42:31.3903637Z printed: x + (x..x)
2019-09-05T21:42:31.3903692Z printed: x + &x
2019-09-05T21:42:31.3903731Z printed: x + (return)
2019-09-05T21:42:31.3904225Z printed: x + (return x)
2019-09-05T21:42:31.3929721Z printed: x + S{..x}
2019-09-05T21:42:31.3929780Z printed: x + x?
2019-09-05T21:42:31.3929820Z printed: x + let _ = x
2019-09-05T21:42:31.3929878Z printed: box x * x
2019-09-05T21:42:31.3929917Z printed: x() * x
2019-09-05T21:42:31.3930150Z printed: x.x(x) * x
2019-09-05T21:42:31.3930189Z printed: x.x(x) * x
2019-09-05T21:42:31.3930244Z printed: (x + x) * x
2019-09-05T21:42:31.3930283Z printed: (x + x) * x
2019-09-05T21:42:31.3930322Z printed: x * x * x
2019-09-05T21:42:31.3930378Z printed: x * x * x
2019-09-05T21:42:31.3930418Z printed: (x << x) * x
2019-09-05T21:42:31.3930528Z printed: (x << x) * x
2019-09-05T21:42:31.3930573Z printed: (x && x) * x
2019-09-05T21:42:31.3930627Z printed: (x && x) * x
2019-09-05T21:42:31.3930666Z printed: (x || x) * x
2019-09-05T21:42:31.3930926Z printed: (x || x) * x
2019-09-05T21:42:31.3930983Z printed: (x < x) * x
2019-09-05T21:42:31.3931024Z printed: (x < x) * x
2019-09-05T21:42:31.3931063Z printed: *x * x
2019-09-05T21:42:31.3931118Z printed: if x { } * x
2019-09-05T21:42:31.3931158Z printed: (move || x) * x
2019-09-05T21:42:31.3931199Z printed: (x = x) * x
2019-09-05T21:42:31.3931239Z printed: (x = x) * x
2019-09-05T21:42:31.3931293Z printed: x.f * x
2019-09-05T21:42:31.3931334Z printed: (x..x) * x
2019-09-05T21:42:31.3931383Z printed: (x..x) * x
2019-09-05T21:42:31.3931443Z printed: &x * x
2019-09-05T21:42:31.3931484Z printed: (return) * x
2019-09-05T21:42:31.3931526Z printed: (return x) * x
2019-09-05T21:42:31.3931567Z printed: S{..x} * x
2019-09-05T21:42:31.3931624Z printed: x? * x
2019-09-05T21:42:31.3931665Z printed: (let _ = x) * x
2019-09-05T21:42:31.3931713Z printed: x * box x
2019-09-05T21:42:31.3931771Z printed: x * x()
2019-09-05T21:42:31.3931811Z printed: x * x.x(x)
2019-09-05T21:42:31.3931850Z printed: x * x.x(x)
2019-09-05T21:42:31.3931910Z printed: x * (x + x)
2019-09-05T21:42:31.3931951Z printed: x * (x + x)
2019-09-05T21:42:31.3931991Z printed: x * (x * x)
2019-09-05T21:42:31.3932031Z printed: x * (x * x)
2019-09-05T21:42:31.3932088Z printed: x * (x << x)
2019-09-05T21:42:31.3932128Z printed: x * (x << x)
2019-09-05T21:42:31.3932169Z printed: x * (x && x)
2019-09-05T21:42:31.3932225Z printed: x * (x && x)
2019-09-05T21:42:31.3932266Z printed: x * (x || x)
2019-09-05T21:42:31.3932312Z printed: x * (x || x)
2019-09-05T21:42:31.3932353Z printed: x * (x < x)
2019-09-05T21:42:31.3932409Z printed: x * (x < x)
2019-09-05T21:42:31.3932450Z printed: x * *x
2019-09-05T21:42:31.3932490Z printed: x * if x { }
2019-09-05T21:42:31.3932545Z printed: x * (move || x)
2019-09-05T21:42:31.3932587Z printed: x * (x = x)
2019-09-05T21:42:31.3932634Z printed: x * (x = x)
2019-09-05T21:42:31.3932691Z printed: x * x.f
2019-09-05T21:42:31.3932732Z printed: x * (x..x)
2019-09-05T21:42:31.3932772Z printed: x * (x..x)
2019-09-05T21:42:31.3932812Z printed: x * &x
2019-09-05T21:42:31.3932869Z printed: x * (return)
2019-09-05T21:42:31.3932909Z printed: x * (return x)
2019-09-05T21:42:31.3932950Z printed: x * S{..x}
2019-09-05T21:42:31.3933004Z printed: x * x?
2019-09-05T21:42:31.3933045Z printed: x * let _ = x
2019-09-05T21:42:31.3933086Z printed: box x << x
2019-09-05T21:42:31.3933127Z printed: x() << x
2019-09-05T21:42:31.3933177Z printed: x.x(x) << x
2019-09-05T21:42:31.3933218Z printed: x.x(x) << x
2019-09-05T21:42:31.3933266Z printed: x + x << x
2019-09-05T21:42:31.3933323Z printed: x + x << x
2019-09-05T21:42:31.3933363Z printed: x * x << x
2019-09-05T21:42:31.3933624Z printed: x * x << x
2019-09-05T21:42:31.3933685Z printed: x << x << x
2019-09-05T21:42:31.3933727Z printed: x << x << x
2019-09-05T21:42:31.3933768Z printed: (x && x) << x
2019-09-05T21:42:31.3934012Z printed: (x && x) << x
2019-09-05T21:42:31.3934089Z printed: (x || x) << x
2019-09-05T21:42:31.3934130Z printed: (x || x) << x
2019-09-05T21:42:31.3934171Z printed: (x < x) << x
2019-09-05T21:42:31.3934230Z printed: (x < x) << x
2019-09-05T21:42:31.3934271Z printed: *x << x
2019-09-05T21:42:31.3934312Z printed: if x { } << x
2019-09-05T21:42:31.3934368Z printed: (move || x) << x
2019-09-05T21:42:31.3934410Z printed: (x = x) << x
2019-09-05T21:42:31.3934452Z printed: (x = x) << x
2019-09-05T21:42:31.3934493Z printed: x.f << x
2019-09-05T21:42:31.3934549Z printed: (x..x) << x
2019-09-05T21:42:31.3934589Z printed: (x..x) << x
2019-09-05T21:42:31.3934730Z printed: &x << x
2019-09-05T21:42:31.3934787Z printed: (return) << x
2019-09-05T21:42:31.3934830Z printed: (return x) << x
2019-09-05T21:42:31.3934873Z printed: S{..x} << x
2019-09-05T21:42:31.3934915Z printed: x? << x
2019-09-05T21:42:31.3934976Z printed: (let _ = x) << x
2019-09-05T21:42:31.3935018Z printed: x << box x
2019-09-05T21:42:31.3935120Z printed: x << x()
2019-09-05T21:42:31.3935185Z printed: x << x.x(x)
2019-09-05T21:42:31.3935227Z printed: x << x.x(x)
2019-09-05T21:42:31.3935270Z printed: x << x + x
2019-09-05T21:42:31.3935329Z printed: x << x + x
2019-09-05T21:42:31.3935371Z printed: x << x * x
2019-09-05T21:42:31.3935413Z printed: x << x * x
2019-09-05T21:42:31.3935455Z printed: x << (x << x)
2019-09-05T21:42:31.3935515Z printed: x << (x << x)
2019-09-05T21:42:31.3935557Z printed: x << (x && x)
2019-09-05T21:42:31.3935598Z printed: x << (x && x)
2019-09-05T21:42:31.3935661Z printed: x << (x || x)
2019-09-05T21:42:31.3935703Z printed: x << (x || x)
2019-09-05T21:42:31.3935746Z printed: x << (x < x)
2019-09-05T21:42:31.3935795Z printed: x << (x < x)
2019-09-05T21:42:31.3935855Z printed: x << *x
2019-09-05T21:42:31.3935897Z printed: x << if x { }
2019-09-05T21:42:31.3935940Z printed: x << (move || x)
2019-09-05T21:42:31.3936001Z printed: x << (x = x)
2019-09-05T21:42:31.3936042Z printed: x << (x = x)
2019-09-05T21:42:31.3936091Z printed: x << x.f
2019-09-05T21:42:31.3936152Z printed: x << (x..x)
2019-09-05T21:42:31.3936194Z printed: x << (x..x)
2019-09-05T21:42:31.3936237Z printed: x << &x
2019-09-05T21:42:31.3936279Z printed: x << (return)
2019-09-05T21:42:31.3936337Z printed: x << (return x)
2019-09-05T21:42:31.3936380Z printed: x << S{..x}
2019-09-05T21:42:31.3936422Z printed: x << x?
2019-09-05T21:42:31.3936482Z printed: x << let _ = x
2019-09-05T21:42:31.3936524Z printed: box x && x
2019-09-05T21:42:31.3936565Z printed: x() && x
2019-09-05T21:42:31.3936607Z printed: x.x(x) && x
2019-09-05T21:42:31.3936666Z printed: x.x(x) && x
2019-09-05T21:42:31.3936708Z printed: x + x && x
2019-09-05T21:42:31.3936757Z printed: x + x && x
2019-09-05T21:42:31.3936817Z printed: x * x && x
2019-09-05T21:42:31.3936858Z printed: x * x && x
2019-09-05T21:42:31.3936901Z printed: x << x && x
2019-09-05T21:42:31.3936960Z printed: x << x && x
2019-09-05T21:42:31.3937002Z printed: x && x && x
2019-09-05T21:42:31.3937044Z printed: x && x && x
2019-09-05T21:42:31.3937093Z printed: (x || x) && x
2019-09-05T21:42:31.3937153Z printed: (x || x) && x
2019-09-05T21:42:31.3937195Z printed: x < x && x
2019-09-05T21:42:31.3937237Z printed: x < x && x
2019-09-05T21:42:31.3937296Z printed: *x && x
2019-09-05T21:42:31.3937338Z printed: if x { } && x
2019-09-05T21:42:31.3937381Z printed: (move || x) && x
2019-09-05T21:42:31.3937423Z printed: (x = x) && x
2019-09-05T21:42:31.3937483Z printed: (x = x) && x
2019-09-05T21:42:31.3937525Z printed: x.f && x
2019-09-05T21:42:31.3937568Z printed: (x..x) && x
2019-09-05T21:42:31.3937628Z printed: (x..x) && x
2019-09-05T21:42:31.3937669Z printed: &x && x
2019-09-05T21:42:31.3937711Z printed: (return) && x
2019-09-05T21:42:31.3937779Z printed: (return x) && x
2019-09-05T21:42:31.3937821Z printed: S{..x} && x
2019-09-05T21:42:31.3937863Z printed: x? && x
2019-09-05T21:42:31.3937905Z printed: let _ = x && x
2019-09-05T21:42:31.3937965Z printed: x && box x
2019-09-05T21:42:31.3938006Z printed: x && x()
2019-09-05T21:42:31.3938216Z printed: x && x.x(x)
2019-09-05T21:42:31.3938275Z printed: x && x.x(x)
2019-09-05T21:42:31.3938315Z printed: x && x + x
2019-09-05T21:42:31.3938355Z printed: x && x + x
2019-09-05T21:42:31.3938394Z printed: x && x * x
2019-09-05T21:42:31.3938673Z printed: x && x * x
2019-09-05T21:42:31.3938715Z printed: x && x << x
2019-09-05T21:42:31.3938918Z printed: x && x << x
2019-09-05T21:42:31.3938976Z printed: x && (x && x)
2019-09-05T21:42:31.3939016Z printed: x && (x && x)
2019-09-05T21:42:31.3939056Z printed: x && (x || x)
2019-09-05T21:42:31.3939112Z printed: x && (x || x)
2019-09-05T21:42:31.3939152Z printed: x && x < x
2019-09-05T21:42:31.3939191Z printed: x && x < x
2019-09-05T21:42:31.3939309Z printed: x && *x
2019-09-05T21:42:31.3939365Z printed: x && if x { }
2019-09-05T21:42:31.3939406Z printed: x && (move || x)
2019-09-05T21:42:31.3939447Z printed: x && (x = x)
2019-09-05T21:42:31.3939503Z printed: x && (x = x)
2019-09-05T21:42:31.3939544Z printed: x && x.f
2019-09-05T21:42:31.3939584Z printed: x && (x..x)
2019-09-05T21:42:31.3939680Z printed: x && (x..x)
2019-09-05T21:42:31.3939744Z printed: x && &x
2019-09-05T21:42:31.3939784Z printed: x && (return)
2019-09-05T21:42:31.3939825Z printed: x && (return x)
2019-09-05T21:42:31.3939884Z printed: x && S{..x}
2019-09-05T21:42:31.3939924Z printed: x && x?
2019-09-05T21:42:31.3939965Z printed: x && let _ = x
2019-09-05T21:42:31.3940022Z printed: box x || x
2019-09-05T21:42:31.3940062Z printed: x() || x
2019-09-05T21:42:31.3940103Z printed: x.x(x) || x
2019-09-05T21:42:31.3940143Z printed: x.x(x) || x
2019-09-05T21:42:31.3940201Z printed: x + x || x
2019-09-05T21:42:31.3940241Z printed: x + x || x
2019-09-05T21:42:31.3940281Z printed: x * x || x
2019-09-05T21:42:31.3940346Z printed: x * x || x
2019-09-05T21:42:31.3940386Z printed: x << x || x
2019-09-05T21:42:31.3940427Z printed: x << x || x
2019-09-05T21:42:31.3940485Z printed: x && x || x
2019-09-05T21:42:31.3940526Z printed: x && x || x
2019-09-05T21:42:31.3940567Z printed: x || x || x
2019-09-05T21:42:31.3940608Z printed: x || x || x
2019-09-05T21:42:31.3940673Z printed: x < x || x
2019-09-05T21:42:31.3940714Z printed: x < x || x
2019-09-05T21:42:31.3940755Z printed: *x || x
2019-09-05T21:42:31.3940813Z printed: if x { } || x
2019-09-05T21:42:31.3940855Z printed: (move || x) || x
2019-09-05T21:42:31.3940896Z printed: (x = x) || x
2019-09-05T21:42:31.3940936Z printed: (x = x) || x
2019-09-05T21:42:31.3940994Z printed: x.f || x
2019-09-05T21:42:31.3941034Z printed: (x..x) || x
2019-09-05T21:42:31.3941074Z printed: (x..x) || x
2019-09-05T21:42:31.3941132Z printed: &x || x
2019-09-05T21:42:31.3941173Z printed: (return) || x
2019-09-05T21:42:31.3941214Z printed: (return x) || x
2019-09-05T21:42:31.3941279Z printed: S{..x} || x
2019-09-05T21:42:31.3941319Z printed: x? || x
2019-09-05T21:42:31.3941360Z printed: let _ = x || x
2019-09-05T21:42:31.3941400Z printed: x || box x
2019-09-05T21:42:31.3941461Z printed: x || x()
2019-09-05T21:42:31.3941502Z printed: x || x.x(x)
2019-09-05T21:42:31.3941542Z printed: x || x.x(x)
2019-09-05T21:42:31.3941607Z printed: x || x + x
2019-09-05T21:42:31.3941648Z printed: x || x + x
2019-09-05T21:42:31.3941689Z printed: x || x * x
2019-09-05T21:42:31.3941728Z printed: x || x * x
2019-09-05T21:42:31.3941792Z printed: x || x << x
2019-09-05T21:42:31.3941832Z printed: x || x << x
2019-09-05T21:42:31.3941872Z printed: x || x && x
2019-09-05T21:42:31.3941929Z printed: x || x && x
2019-09-05T21:42:31.3941970Z printed: x || (x || x)
2019-09-05T21:42:31.3942011Z printed: x || (x || x)
2019-09-05T21:42:31.3942069Z printed: x || x < x
2019-09-05T21:42:31.3942109Z printed: x || x < x
2019-09-05T21:42:31.3942149Z printed: x || *x
2019-09-05T21:42:31.3942190Z printed: x || if x { }
2019-09-05T21:42:31.3942254Z printed: x || (move || x)
2019-09-05T21:42:31.3942296Z printed: x || (x = x)
2019-09-05T21:42:31.3942336Z printed: x || (x = x)
2019-09-05T21:42:31.3942393Z printed: x || x.f
2019-09-05T21:42:31.3942434Z printed: x || (x..x)
2019-09-05T21:42:31.3942474Z printed: x || (x..x)
2019-09-05T21:42:31.3942515Z printed: x || &x
2019-09-05T21:42:31.3942582Z printed: x || (return)
2019-09-05T21:42:31.3942624Z printed: x || (return x)
2019-09-05T21:42:31.3942664Z printed: x || S{..x}
2019-09-05T21:42:31.3942721Z printed: x || x?
2019-09-05T21:42:31.3942762Z printed: x || let _ = x
2019-09-05T21:42:31.3942802Z printed: box x < x
2019-09-05T21:42:31.3942860Z printed: x() < x
2019-09-05T21:42:31.3942901Z printed: x.x(x) < x
2019-09-05T21:42:31.3942941Z printed: x.x(x) < x
2019-09-05T21:42:31.3942982Z printed: x + x < x
2019-09-05T21:42:31.3943040Z printed: x + x < x
2019-09-05T21:42:31.3943081Z printed: x * x < x
2019-09-05T21:42:31.3943121Z printed: x * x < x
2019-09-05T21:42:31.3943253Z printed: x << x < x
2019-09-05T21:42:31.3943294Z printed: x << x < x
2019-09-05T21:42:31.3943334Z printed: (x && x) < x
2019-09-05T21:42:31.3943375Z printed: (x && x) < x
2019-09-05T21:42:31.3943433Z printed: (x || x) < x
2019-09-05T21:42:31.3943473Z printed: (x || x) < x
2019-09-05T21:42:31.3943514Z printed: x < x < x
2019-09-05T21:42:31.3944153Z ------------------------------------------
2019-09-05T21:42:31.3944210Z stderr:
2019-09-05T21:42:31.3944461Z ------------------------------------------
2019-09-05T21:42:31.3944461Z ------------------------------------------
2019-09-05T21:42:31.3945829Z thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Diagnostic { level: Error, message: [("chained comparison operators require parentheses", NoStyle)], code: None, span: MultiSpan { primary_spans: [Span { lo: BytePos(3490), hi: BytePos(3497), ctxt: #0 }], span_labels: [] }, children: [SubDiagnostic { level: Help, message: [("use `::<...>` instead of `<...>` if you meant to specify type arguments", NoStyle)], span: MultiSpan { primary_spans: [], span_labels: [] }, render_span: None }, SubDiagnostic { level: Help, message: [("or use `(...)` if you meant to specify fn arguments", NoStyle)], span: MultiSpan { primary_spans: [], span_labels: [] }, render_span: None }], suggestions: [] }', src/libcore/result.rs:1165:5
2019-09-05T21:42:31.3946031Z 
2019-09-05T21:42:31.3946273Z ------------------------------------------
2019-09-05T21:42:31.3946306Z 
2019-09-05T21:42:31.3946348Z 
---
2019-09-05T21:42:31.3949103Z :536:22
2019-09-05T21:42:31.3951882Z note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
2019-09-05T21:42:31.3952050Z 
2019-09-05T21:42:31.3952103Z 
2019-09-05T21:42:31.3953744Z command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui-fulldeps" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-6.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet" "--llvm-version" "6.0.0\n" "--system-llvm" "--cc" "" "--cxx" "" "--cflags" "" "--llvm-components" "" "--llvm-cxxflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"
2019-09-05T21:42:31.3954765Z 
2019-09-05T21:42:31.3954821Z 
2019-09-05T21:42:31.3954871Z failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test
2019-09-05T21:42:31.3954967Z Build completed unsuccessfully in 1:15:53
2019-09-05T21:42:31.3954967Z Build completed unsuccessfully in 1:15:53
2019-09-05T21:42:31.3970503Z == clock drift check ==
2019-09-05T21:42:31.3988289Z   local time: Thu Sep  5 21:42:31 UTC 2019
2019-09-05T21:42:31.5507183Z   network time: Thu, 05 Sep 2019 21:42:31 GMT
2019-09-05T21:42:31.5507315Z == end clock drift check ==
2019-09-05T21:42:32.1994380Z ##[error]Bash exited with code '1'.
2019-09-05T21:42:32.2044968Z ##[section]Starting: Checkout
2019-09-05T21:42:32.2046851Z ==============================================================================
2019-09-05T21:42:32.2046906Z Task         : Get sources
2019-09-05T21:42:32.2046971Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

--- a/src/test/ui-fulldeps/pprust-expr-roundtrip.rs
+++ b/src/test/ui-fulldeps/pprust-expr-roundtrip.rs
@@ -32,13 +32,15 @@ use syntax::print::pprust;
 use syntax::ptr::P;
 
 
-fn parse_expr(ps: &ParseSess, src: &str) -> P<Expr> {
+fn parse_expr(ps: &ParseSess, src: &str) -> Option<P<Expr>> {
     let src_as_string = src.to_string();
 
-    let mut p = parse::new_parser_from_source_str(ps,
-                                                  FileName::Custom(src_as_string.clone()),
-                                                  src_as_string);
-    p.parse_expr().unwrap()
+    let mut p = parse::new_parser_from_source_str(
+        ps,
+        FileName::Custom(src_as_string.clone()),
+        src_as_string,
+    );
+    p.parse_expr().map_err(|mut e| e.cancel()).ok()
 }
 
 
@@ -209,22 +211,23 @@ fn run() {
         let printed = pprust::expr_to_string(&e);
         println!("printed: {}", printed);
 
-        let mut parsed = parse_expr(&ps, &printed);
-
-        // We want to know if `parsed` is structurally identical to `e`, ignoring trivial
-        // differences like placement of `Paren`s or the exact ranges of node spans.
-        // Unfortunately, there is no easy way to make this comparison. Instead, we add `Paren`s
-        // everywhere we can, then pretty-print. This should give an unambiguous representation of
-        // each `Expr`, and it bypasses nearly all of the parenthesization logic, so we aren't
-        // relying on the correctness of the very thing we're testing.
-        RemoveParens.visit_expr(&mut e);
-        AddParens.visit_expr(&mut e);
-        let text1 = pprust::expr_to_string(&e);
-        RemoveParens.visit_expr(&mut parsed);
-        AddParens.visit_expr(&mut parsed);
-        let text2 = pprust::expr_to_string(&parsed);
-        assert!(text1 == text2,
-                "exprs are not equal:\n  e =      {:?}\n  parsed = {:?}",
-                text1, text2);
+        // Ignore expressions with chained comparisons that fail to parse
+        if let Some(mut parsed) = parse_expr(&ps, &printed) {
+            // We want to know if `parsed` is structurally identical to `e`, ignoring trivial
+            // differences like placement of `Paren`s or the exact ranges of node spans.
+            // Unfortunately, there is no easy way to make this comparison. Instead, we add `Paren`s
+            // everywhere we can, then pretty-print. This should give an unambiguous representation
+            // of each `Expr`, and it bypasses nearly all of the parenthesization logic, so we
+            // aren't relying on the correctness of the very thing we're testing.
+            RemoveParens.visit_expr(&mut e);
+            AddParens.visit_expr(&mut e);
+            let text1 = pprust::expr_to_string(&e);
+            RemoveParens.visit_expr(&mut parsed);
+            AddParens.visit_expr(&mut parsed);
+            let text2 = pprust::expr_to_string(&parsed);
+            assert!(text1 == text2,
+                    "exprs are not equal:\n  e =      {:?}\n  parsed = {:?}",
+                    text1, text2);
+        }
     });
 }

